### DISCUSSION TITLE
How do switch Python deps to monthly update?

### DISCUSSION THREAD

**Discussion by khitrenovich** [renovatebot/renovate#12679] [Request Help]
We are using Renovate for relatively long time now, and are very happy about it. 

We have a monorepo, which is mostly NodeJS-based. Recently we've added a Python script to one of the modules and were surprised to see the Renovate picked Python deps too! Unfortunately, one of the deps (namely, `boto3` is being updated pretty much daily, which creates unnecessary noise for us - after all, it is just a helper dev-time script. 

I've tried to switch Python to monthly updates with the config below in the `packageRules`, but it did not work - we keep getting fresh `boto3` PRs.

```
    {
      "matchDatasources": ["python"],
      "extends": [ "schedule:monthly" ]
    },
```

What did I miss? Thanks in advance!

**Comment by viceice** [comment]
Wrong datasource name. 😉

https://docs.renovatebot.com/modules/datasource/

You probably need `pypi` datasource.

**Reply by khitrenovich** [reply]
Thank you, I'll give it a try shortly!

Unfortunately, I hardly know anything about Python ecosystem... I saw `python` being mentioned in https://docs.renovatebot.com/python/#disabling-python-support and took it as a hint. I guess I was wrong :)

**Comment by HonkingGoose** [comment]
**Edit:** Add `matchDatasources` = `pypi` to config proposal.

---

Alternative option: tell Renovate to update just your noisy `boto3` Python package on a schedule [^schedule], but get other Python package updates right away:

```json
{
  "packageRules": [
    {
      "matchDatasources": ["pypi"],
      "matchPackageNames": ["boto3"],
      "schedule": ["before 3am on the first day of the month"]
    }
  ]
}
```

I recommend you read the Noise Reduction documentation [^noise] as well. 😉 

[^schedule]: https://docs.renovatebot.com/configuration-options/#schedule
[^noise]: https://docs.renovatebot.com/noise-reduction/

**Reply by khitrenovich** [reply]
Thanks @HonkingGoose, we are using this already extensively for `npm` (yes, for `aws-sdk` packages too 😉 ).
